contracts/modules/licensing/BaseLicenseTemplate.sol

```diff
@@ -0,0 +1,56 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.23;
+
+// external
+import { ERC165 } from "@openzeppelin/contracts/utils/introspection/ERC165.sol";
+import { IERC165 } from "@openzeppelin/contracts/utils/introspection/IERC165.sol";
+import { Initializable } from "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+// contracts
+import { ILicenseTemplate } from "../../interfaces/modules/licensing/ILicenseTemplate.sol";
+
+abstract contract BaseLicenseTemplate is ILicenseTemplate, ERC165, Initializable {
+    /// @custom:storage-location erc7201:story-protocol.BaseLicenseTemplate
+    struct BaseLicenseTemplateStorage {
```

Comment:
By making this contract `Initializable`, we are heavily indicating to devs writing other Templates that they must use upgradeability. I think we should be neutral.

Options:
- Renaming `BaseLicenseTemplate` to `BaseLicenseUpgradeable`, so we can have a BaseLicenseTemplate version later on.
- Removing `Initializable` and making `name` and `metadataUri` immutable. This would make BaseLicenseTemplate safe for upgradeable and non upgradeable contracts since it won't have storage slots. There is no setter for `name` and `metadataURI` in this contract, so it seems like we already don't thing Template devs changing them later on is desired.

This is a business decision IMO. Upgradeable contracts can change these parameters even without setters, but we should be clear with our expectations for devs.

Reply:
Make sense! Renamed to `BaseLicenseTemplateUpgradeable`